WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -29,75 +29,114 @@ class RegisterDomainCoordinator {
-                               url: url,
-                               title: nil,
-                               shouldPush: false)
+                self.domainAddedToCartCallback?(viewController, domain.domainName, blog)
+                onSuccess()
+            },
+            onFailure: onFailure
+        )
     }
 
+    /// Related to the `purchaseFromDomainManagement` Domain selection type.
+    /// Adds the selected domain to the cart then launches the checkout webview
+    /// The checkout webview is configured for the domain management flow
     func handleNoSiteChoice(on viewController: UIViewController) {
         createCart(
             onSuccess: { [weak self] in
-                self?.presentWebViewForNoSite(on: viewController)
+                self?.presentCheckoutWebview(on: viewController, title: TextContent.checkoutTitle, shouldPush: true)
             }) {
                 viewController.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
             }
     }
 
+    /// Related to the `purchaseFromDomainManagement` Domain selection type.
+    /// Adds the selected domain to the cart then presents a site picker view.
     func handleExistingSiteChoice(on viewController: UIViewController) {
-        print("handleExistingSiteChoice")
+        let config = BlogListConfiguration(shouldShowCancelButton: false,
+                                           shouldShowNavBarButtons: false,
+                                           navigationTitle: TextContent.sitePickerNavigationTitle,
+                                           backButtonTitle: TextContent.sitePickerNavigationTitle,
+                                           shouldHideSelfHostedSites: true)
+        guard let blogListViewController = BlogListViewController(configuration: config, meScenePresenter: nil) else {
+            return
+        }
+
+        blogListViewController.blogSelected = { [weak self] controller, selectedBlog in
+            guard let self,
+                  let blog = selectedBlog else {
+                return
+            }
+            self.site = blog
+            controller?.showLoading()
+            self.createCart {
+                if let controller,
+                   let domainName = self.domain?.domainName {
+                    self.domainAddedToCartCallback?(controller, domainName, blog)
+                }
+                controller?.hideLoading()
+            } onFailure: {
+                controller?.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
+                controller?.hideLoading()
+            }
+
+        }
+
+        viewController.navigationController?.pushViewController(blogListViewController, animated: true)
     }
 
     // MARK: Helpers
 
+    private func createCart(onSuccess: @escaping () -> (),
+                    onFailure: @escaping () -> ()) {
+        guard let domain else { return }
```

Comment:
[Suggestion] If `domain` is `nil`, then this method will never resolve and its callers are stuck forever. the `onFailure` block should be called with an appropriate error.

Reply:
Result makes sense

Reply:
Thanks for the `Result` suggestion. Addressed in caa3bb7e2127613bea0e9ba624e295066d540dec 👍